File: fluentcore/src/main/java/com/mcreater/fxfluent/controls/state/StateUtil.java
package com.mcreater.fxfluent.controls.state;

public class StateUtil {
    public static StateType genState(boolean disabled, boolean hovered, boolean pressed, boolean focused) {
        return disabled ? StateType.DISABLED : (pressed ? StateType.PRESSED : (hovered ? StateType.HOVER : (focused ? StateType.FOCUS : StateType.NONE)));
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/controls/state/StateUtil.kt
package com.mcreater.fxfluent.controls.state

class StateUtil {
    companion object {
        @JvmStatic
        fun genState(disabled: Boolean, hovered: Boolean, pressed: Boolean, focused: Boolean): StateType {
            return if (disabled) StateType.DISABLE else if (pressed) StateType.PRESS else if (hovered) StateType.HOVER else if (focused) StateType.FOCUS else StateType.NONE
        }
    }
}